pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/inverted/InvertedIndexType.java

```diff
@@ -193,4 +196,16 @@ public InvertedIndexReader createSkippingForward(SegmentDirectory.Reader segment
   protected void handleIndexSpecificCleanup(TableConfig tableConfig) {
     tableConfig.getIndexingConfig().setInvertedIndexColumns(null);
   }
+
+  @Nullable
+  @Override
+  public MutableIndex createMutableIndex(MutableIndexContext context, IndexConfig config) {
+    if (config.isDisabled()) {
+      return null;
+    }
+    if (!context.hasDictionary()) {
+      return null;
```

Comment:
Shouldn't we be throwing IllegalStateException here?

Reply:
Other places too. I think a lot of the cases are table creation level validations.

Reply:
I guess this is similar to what Neha asked [here](https://github.com/apache/pinot/pull/10687#discussion_r1183142857).

The TL;DR: is that in that situation previous code didn't fail but simply ignore the index, so that is what we do here.

There is a difference, though: In previous code the MutableIndex was created and stored, but no row was ever added to it. With this version the MutableIndex is never created. I'm assuming both behaviors are compatible. Queries that theoretically could use the index, should be skipping it in the older case or otherwise they would return invalid data (because it is actually empty)